### PR TITLE
Add information for build system dependencies

### PR DESCRIPTION
**Pull request by jeffaco**
@Microsoft/ostc-devs @OpusDude 

Steve, please check O/S versions (columns, in particular), I'm not 100% sure I have the nomenclature correct. Thanks!

### CODE REVIEW

File: setup-builddeps.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+### Solaris 10 Sparc
+
+Component | Version | Revision
+--------- | ------- | --------
+Release | Oracle Solaris 10 1/13 | s10s_u11wos_24a SPARC
+SUNWlibC | 5.10 | REV-2004.12.22
+SUNWlibmsr | 5.10 | REV=2004.11.23 
+SUNWcslr | 11.10.0 | REV=2005.01.21.15.53
+SUNWcsl | 11.10.0 | REV=2005.01.21.15.53
+SUNWcsr -  11.10.0, REV=2005.01.21.15.53
```
Comment (niroyb) [inline setup-builddeps.md]: Little table formatting bug here

File: setup-builddeps.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+SUNWcslr | 11.10.0 | REV=2005.01.21.16.34
+SUNWcsl | 11.10.0 | REV=2005.01.21.16.34
+SUNWcsr | 11.10.0 | REV=2005.01.21.16.34
+SUNWopenssl-libraries | 11.10.0 | REV=2005.01.21.16.34
+
+### Solaris 11 Sparc
+
+Component | Version | Revision
+--------- | ------- | --------
+Release | Oracle Solaris 11 11/11 SPARC
```
Comment (niroyb) [inline setup-builddeps.md:81]: Missing release revision?

**Review by niroyb — APPROVED** [review]

File: setup-builddeps.md (hunk 1 above)
Comment (jeffaco) [reply setup-builddeps.md]: Caught that, fixed.

File: setup-builddeps.md (hunk 2 above)
Comment (jeffaco) [reply setup-builddeps.md:81]: Not 100% sure what to put here, SteveW will tell me.

Comment (OpusDude) [reply setup-builddeps.md:81]: Solaris 11 SPARC or X86 does not have a release revision.

**Review by OpusDude — APPROVED** [review]
Looks good

**Pull request merged**